### PR TITLE
Sending resource usage stats in ReportComputedTask

### LINKED ISSUE #4526: Provider usage reporting for gWASM subtasks (GR-183)
## Description

US. As a Golem requestor I need to get usage  report for every subtask completed in order to calculate payment and evaluate provider according to my market strategy.

It includes change in RCT message. It can be extra data. (possibly it should be a separate field?)

As soon as a subtask verification is finished, usages of providers who passed verification should be reported to the market strategy module.

AC. PR sent to review.

### PR DESCRIPTION
**Pull request by kmazurek**
Resolves: #4526

Adds the logic for extracting usage statistics and including them in the
ReportComputedTask message. The stats are extracted from a file created
by Docker jobs in a mounted volume.

### CODE REVIEW

**Comment by mplebanski** [comment]
We could use this opportunity to make WaitingTaskResult a dataclass, what do you think?

**Comment by kmazurek** [comment]
@mplebanski Done, I've converted `WaitingTaskResult` and `WaitingTaskFailure` to data classes.

File: golem/docker/task_thread.py [hunk 1]
```diff
@@ -238,6 +240,19 @@ def get_progress(self):
...
+        stats_file: Path = self.dir_mapping.stats / DockerJob.STATS_FILE
+
+        if not stats_file.exists():
+            return {}
+
+        try:
+            with stats_file.open() as f:
+                return json.load(f)
+        except json.JSONDecodeError as e:
+            logger.warning(f'Failed to parse stats file: {stats_file}. {e}')
```
Comment (Wiezzel) [inline golem/docker/task_thread.py]:
```suggestion
            logger.warning(f'Failed to parse stats file: {stats_file}.', exc_info=e)
```

File: golem/task/taskserver.py [hunk 2]
```diff
@@ -1057,36 +1063,29 @@ def __get_task_manager_root(datadir):
...
+
+    already_sending: bool = False
+    package_path: Optional[str] = None
+    package_sha1: Optional[str] = None
+    result_hash: Optional[str] = None
+    result_path: Optional[str] = None
+    result_secret: Optional[str] = None
+    result_sha1: Optional[str] = None
+    result_size: int = 0
+    stats: Dict = field(default_factory=lambda: {})
```
Comment (Wiezzel) [inline golem/task/taskserver.py]:
```suggestion
    stats: Dict = field(default_factory=dict)
```

**Review by Wiezzel — APPROVED** [review]
LGTM
Two cosmetic suggestions.

**Comment by mbenke** [comment]
Seems to fail VbR integration tests: https://buildbot.golem.network/buildbot/#/builders/15/builds/875/

**Pull request merged**